framework/src/play-filters-helpers/src/test/scala/play/filters/csrf/CSRFFilterSpec.scala

```diff
@@ -41,11 +40,29 @@ class CSRFFilterSpec extends CSRFCommonSpecs {
     "not add a token to GET requests that don't accept HTML" in {
       buildCsrfAddToken()(_.addHttpHeaders(ACCEPT -> "application/json").get())(_.status must_== NOT_FOUND)
     }
-    "not add a token to responses that set cache headers" in {
-      buildCsrfAddResponseHeaders(CACHE_CONTROL -> "public, max-age=3600")(_.get())(_.cookies must be empty)
+    "not add a token to non GET request when response is cached" in {
+      Fragment.foreach(Seq("POST", "PUT", "DELETE")) { method =>
+        method >> {
+          buildCsrfAddResponseHeaders(CACHE_CONTROL -> "public, max-age=2600")(_.execute(method))(_.cookies must be empty)
+        }
+      }
+    }
+    "not add a token to non GET request when response is not cached " in {
+      Fragment.foreach(Seq("POST", "PUT", "DELETE")) { method =>
+        method >> {
+          buildCsrfAddResponseHeaders()(_.addHttpHeaders(CACHE_CONTROL -> "no-cache").execute(method))(_.cookies must be empty)
+        }
+      }
+    }
+    "add a token to GET request when response is not cached" in {
+      Fragment.foreach(Seq("no-cache", "no-store", "NO-CACHE", "NO-STORE ", "no-cache, must-revalidate")) { directive =>
```

Comment:
Add test for the `private` directive?

```suggestion
      Fragment.foreach(Seq("no-cache", "no-store", "private", "NO-CACHE", "NO-STORE ", "no-cache, must-revalidate")) { directive =>
```